### PR TITLE
Use esbuild to build js/ts

### LINKED ISSUE #4: Switch to esbuild
Should reduce the time it takes to deploy and run tests by multiple orders of magnitude.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add esbuild-loader
- Add turboLoaded property to Window
- Simplify cast
- Use esbuild for js/ts

Closes https://github.com/exercism/project-management/issues/4

### CODE REVIEW

File: app/javascript/packs/core.tsx [hunk 1]
```diff
@@ -9,6 +9,4 @@ import { navigator } from '@hotwired/turbo'
 // As we're sensitive to the order of things across different packs
 // we set a window-level constant to record when turbo has loaded
 // so that other packs that haven't yet rendered events can respond to them
-document.addEventListener('turbo:load', () => {
-  ;(window as any).turboLoaded = true
-})
+document.addEventListener('turbo:load', () => (window.turboLoaded = true))
```
Comment (ErikSchierboom) [inline app/javascript/packs/core.tsx:12]: The above addition to `declarations.d.ts` makes this change possible.

File: config/webpack/environment.js [hunk 2]
```diff
@@ -13,5 +13,12 @@ environment.plugins.prepend(
   )
 )
 
+environment.loaders.get('babel').use = [
+  { loader: 'esbuild-loader', options: { loader: 'tsx', target: 'es2015' } },
+]
+environment.loaders.get('nodeModules').use = [
+  { loader: 'esbuild-loader', options: { loader: 'jsx', target: 'es2015' } },
+]
```
Comment (ErikSchierboom) [inline config/webpack/environment.js:21]: With webpacker 6, this integration becomes a lot easier.

**Review by ErikSchierboom — COMMENTED** [review]
I found it slightly hard to performance test this, as I don't really know where any caching occured.

**Comment by kntsoriano** [comment]
This makes me really excited!

**Comment by ErikSchierboom** [comment]
Note: I had to change all tests that used React from `.js` to `.jsx`, as otherwise esbuild wouldn't apply any React transformations on the code (which makes sense).

**Comment by iHiD** [comment]
Looks great. I'll merge tomorrow when you're on :)

**Comment by ErikSchierboom** [comment]
@iHiD I've noticed that the code coverage check fails quite regularly. Is this valid or should we make it less strict?

**Comment by iHiD** [comment]
I'm not sure why it fails. It's often that same test.

**Review by kntsoriano — APPROVED** [review]
Looks great :)

**Pull request merged**